apps/api/src/app/events/usecases/send-message/digest.usecase.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@nestjs/common';
+import { JobStatusEnum, MessageRepository } from '@novu/dal';
+import { CreateLog } from '../../../logs/usecases/create-log/create-log.usecase';
+import { SendMessageCommand } from './send-message.command';
+import { SendMessageType } from './send-message-type.usecase';
+import { JobRepository } from '../../../../../../../libs/dal/src/repositories/job/job.repository';
+import moment from 'moment';
+
+@Injectable()
+export class Digest extends SendMessageType {
+  constructor(
+    protected messageRepository: MessageRepository,
+    protected createLogUsecase: CreateLog,
+    protected jobRepository: JobRepository
+  ) {
+    super(messageRepository, createLogUsecase);
+  }
+
+  public async execute(command: SendMessageCommand) {
+    const currentJob = await this.jobRepository.findOne({
+      _id: command.jobId,
+    });
+
+    const earliest = moment(currentJob.createdAt).subtract(10, 'minutes').toDate();
+    const jobs = await this.jobRepository.find({
+      updatedAt: {
```

Comment:
Adding additional fields such as environmentId, subscriberId and etc here is important. 

Maybe extracting this to a function in the jobRepository to control the query in a more type safe manner, since it might be duplicated across multiple areas